server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -577,11 +576,11 @@ protected Node(
         }
     }
 
-    static void warnIfPreRelease(final Version version, final boolean isSnapshot, final Logger logger) {
-        if (!version.isRelease() || isSnapshot) {
+    static void warnIfPreRelease(final Build build,final Logger logger) {
+        if (build.isProductionRelease() == false) {
```

Comment:
I think maybe this isn't worth being its own method any more. Now that we're testing `isProductionRelease` instead.